zerver/tests/test_event_queue.py

```diff
@@ -128,456 +119,421 @@ def test_stream_watchers(self) -> None:
-            stream: Stream,
-            sub: Subscription,
-            properties: Dict[str, bool],
-        ) -> None:
-            for property_name, value in properties.items():
-                do_change_subscription_property(
-                    user_profile, sub, stream, property_name, value, acting_user=None
-                )
-
-        def allocate_event_queue(user: UserProfile) -> ClientDescriptor:
-            result = self.tornado_call(
-                get_events,
-                user,
-                {
-                    "apply_markdown": orjson.dumps(True).decode(),
-                    "client_gravatar": orjson.dumps(True).decode(),
-                    "event_types": orjson.dumps(["message"]).decode(),
-                    "user_client": "website",
-                    "dont_block": orjson.dumps(True).decode(),
-                },
-            )
-            self.assert_json_success(result)
-            queue_id = orjson.loads(result.content)["queue_id"]
-            return access_client_descriptor(user.id, queue_id)
-
-        def destroy_event_queue(user: UserProfile, queue_id: str) -> None:
-            result = self.tornado_call(cleanup_event_queue, user, {"queue_id": queue_id})
-            self.assert_json_success(result)
-
-        def assert_maybe_enqueue_notifications_call_args(
-            args_dict: Collection[Any],
-            message_id: int,
-            user_id: int,
-            **kwargs: Any,
-        ) -> None:
-            expected_args_dict = self.get_maybe_enqueue_notifications_parameters(
-                user_id=user_id,
-                acting_user_id=iago.id,
-                message_id=message_id,
-                **kwargs,
+    def assert_maybe_enqueue_notifications_call_args(
+        self,
+        args_dict: Collection[Any],
+        message_id: int,
+        user_id: int,
+        **kwargs: Any,
+    ) -> None:
+        expected_args_dict = self.get_maybe_enqueue_notifications_parameters(
+            user_id=user_id,
+            acting_user_id=self.example_user("iago").id,
+            message_id=message_id,
+            **kwargs,
+        )
+        self.assertEqual(args_dict, expected_args_dict)
+
+    def change_subscription_properties(
+        self,
+        user_profile: UserProfile,
+        stream: Stream,
+        sub: Subscription,
+        properties: Dict[str, bool],
+    ) -> None:
+        for property_name, value in properties.items():
+            do_change_subscription_property(
+                user_profile, sub, stream, property_name, value, acting_user=None
             )
-            self.assertEqual(args_dict, expected_args_dict)
 
-        client_descriptor = allocate_event_queue(user_profile)
+    def setUp(self) -> None:
+        self.user_profile = self.example_user("hamlet")
+        self.cordelia = self.example_user("cordelia")
+        self.user_profile.enable_online_push_notifications = False
+        self.user_profile.save()
```

Comment:
Could we use the `actions.py` function to modify this property? 

We're generally trying to get rid of test code that modifies objects directly like this, since it can fail to mimic reality correctly.

Reply:
Done

Reply:
There's actually many of these in the codebase:
```shell
(ins)code/zulip> grep -r "\.save()" zerver/tests/ | wc -l
323
```
Would it be possible to change these to use the `/actions` functions in a (semi) automated way?

Reply:
I don't see a super automated way to do it, but it is a sweep that would be well worth doing. There's probably a few cases where it wouldn't work to switch to the `actions` function, where we might want to add a comment explaining why we're doing it manually, so requires some reading/care, but I could imagine doing such a sweep model-by-model (say, starting with `realm.save()`) being a project that'd finish reasonably quickly.